### DISCUSSION TITLE
DefaultValues for checkbox group

### DISCUSSION THREAD

**Discussion by ahmedrafayat** [react-hook-form/react-hook-form#3054] [Q&A] [answered]
I am trying to create a checkbox group with the same name. Currently, if I give all the checkboxes the same name, handleSubmit returns the data in the form of an array like

```
{
  nameOfField: [value1, value2]
}
```

which is expected and this is what I want.
But my problem is I don't know how to assign defaultValues to the checkbox fields. Important Note: I am fetching the fields from an API, so my form is dynamically generated.

An example of what I am trying to do can be seen [here in this sandbox](https://codesandbox.io/s/peaceful-dream-dg1fx?file=/src/App.js)

**Comment by ahmedrafayat** [comment, marked as answer]
I managed to solve this problem earlier by just using React's built in defaultValue/defaultChecked parameter

**Reply by danielzzz** [reply]
hi, do you have a working example of this?
